### ISSUE TITLE
Retrieve data directly from Trafas API and do calculations on our end

### ISSUE THREAD

**Issue by elvbom** [Klimatbyran/klimatkollen#348]
Currently, we get data on the rate of increase of the share of newly registered rechargeable cars in the municipalities from an Excel file (/data/solutions/cars/sources/kpi1_calculations.xlsx) based on statistics from Trafa. However, Trafa has an API we can use and then perform calculations on the data. This is better as it makes it easier to keep the information updated.

Steps to go through:

- [ ] Obtain access to Trafa's API (is a key needed?)
- [ ] Download data on the total number of newly registered cars and the number of newly registered electric cars, on an annual basis
- [ ] Write tests for new functions so that the data from 2015-2022 matches the calculations in the Excel file
- [ ] Implement the new functions

For questions, please contact @elvbom who can explain in detail if needed

**Comment by LudwikJaniuk** [comment]
I propose to update the title to
"Retrieve electric car data directly from Trafas API"

**Comment by LudwikJaniuk** [comment]
@Braggedtooth This was not a trivial issue, so kudos for picking it up during the hackathon. Did you manage to get some parts done? And would you like to continue your efforts in the future? We are here to help if you got stuck on some part of it.

**Comment by Braggedtooth** [comment]
> @Braggedtooth This was not a trivial issue, so kudos for picking it up during the hackathon. Did you manage to get some parts done? And would you like to continue your efforts in the future? We are here to help if you got stuck on some part of it.

Indeed, although i ended up working on something else during the hackathon i do plan to continue working on this issue.
So far after diving into the specifics of the API, I am planning to first develop a wrapper for the API.
However, before proceeding, I'd like to understand if the project might benefit from other aspects of the Trafa API data beyond electric car statistics. 
This will help in determining whether a comprehensive implementation is necessary or if a more focused approach is preferable.
I would also appreciate a more detailed insights on the data and functionality we aim to implement.

**Comment by Braggedtooth** [comment]
Here is my progress so far, please have a look.  @LudwikJaniuk  @elvbom 
https://github.com/Braggedtooth/klimatkollen/blob/fix-%23348/data/trafa/index.ts

Let me know if this is on the right path.

**Comment by LudwikJaniuk** [comment]
I'll have a look this evening unless someone else beats me to it!

**Comment by LudwikJaniuk** [comment]
To update the status, @Braggedtooth and I had a call last month and discussed some ways to move forward.
